### ISSUE TITLE
High preset for a 4k webcam records in 1080p

### ISSUE THREAD

**Issue by RoyHP** [matthutchinson/videosnap#20]
Hi, the high preset for my 4k webcam (Logi 4k Pro) records in 1080p and I don't see a way to record natively in 4k. Any idea about this?

Let me know if I can help - thanks!

**Comment by matthutchinson** [comment]
Sorry I don't have a 4k camera to be able to try this out.

**Comment by matthutchinson** [comment]
I'm validating the chosen encoding preset [here](https://github.com/matthutchinson/videosnap/blob/main/videosnap/videosnap.m#L240) and then set it in the capture session [here](https://github.com/matthutchinson/videosnap/blob/main/videosnap/videosnap.m#L353).

Right now, the preset must [be one of these](https://github.com/matthutchinson/videosnap/blob/main/videosnap/Constants.m#L27). This list was taken from older Apple API docs on capture sessions.  It looks like there are now significantly more options.

https://developer.apple.com/documentation/avfoundation/avcapturesession/preset

When I have time available I can look at introducing all these newer options.

**Comment by matthutchinson** [comment]
😉 Or feel free to submit a PR for this if you know how.

**Comment by RoyHP** [comment]
Thanks - I'm happy to provide the PR.

**Comment by RoyHP** [comment]
Okay so I tried but it gave me "could not set encoding preset '3840x2160' for this device, defaulting to Medium" - I tried some other variations of the preset as well, and got the same message. See attached screenshot.

I'll try to see if OBS can use it at 4k as a sanity check.

<img width="1323" alt="Screen Shot 2021-02-28 at 9 20 57 PM" src="https://user-images.githubusercontent.com/1066439/109444506-643ed380-7a0b-11eb-94ba-22659d705561.png">
